evalml/pipelines/utils.py

```diff
@@ -815,6 +816,75 @@ def make_timeseries_baseline_pipeline(problem_type, gap, forecast_horizon, time_
     return baseline
 
 
+def are_datasets_separated_by_gap_time_index(train, test, pipeline_params):
+    """Determine if the train and test datasets are separated by gap number of units using the time_index.
+
+    This will be true when users are predicting on unseen data but not during cross
+    validation since the target is known.
+
+    Args:
+        train (pd.DataFrame): Training data.
+        test (pd.DataFrame): Data of shape [n_samples, n_features].
+        pipeline_params (dict): Dictionary of time series parameters.
+
+    Returns:
+        bool: True if the difference in time units is equal to gap + 1.
+
+    """
+    gap_difference = pipeline_params["gap"] + 1
+
+    train_copy = train.copy()
+    test_copy = test.copy()
+    train_copy.ww.init(time_index=pipeline_params["time_index"])
+    test_copy.ww.init(time_index=pipeline_params["time_index"])
+
+    X_frequency_dict = train_copy.ww.infer_temporal_frequencies(
+        temporal_columns=[train_copy.ww.time_index]
+    )
+    freq = X_frequency_dict[test_copy.ww.time_index]
+    if freq is None:
+        return True
+
+    first_testing_date = test_copy[test_copy.ww.time_index].iloc[0]
+    last_training_date = train_copy[train_copy.ww.time_index].iloc[-1]
+    dt_difference = first_testing_date - last_training_date
+
+    try:
+        units_difference = dt_difference / freq
+    except ValueError:
+        units_difference = dt_difference / ("1" + freq)
+    return units_difference == gap_difference
+
+
+def validate_holdout_datasets(X, X_train, pipeline_params):
+    """Validate the holdout datasets match out expectations.
+
+    Args:
+        X (pd.DataFrame): Data of shape [n_samples, n_features].
+        X_train (pd.DataFrame): Training data.
+        pipeline_params (dict): Dictionary of time series parameters.
+
+    Raises:
+        PartialDependenceError: If holdout data does not have forecast_horizon entries or if datasets are not separated by gap.
+    """
+    forecast_horizon = pipeline_params["forecast_horizon"]
+    gap = pipeline_params["gap"]
+    time_index = pipeline_params["time_index"]
+    right_length = len(X) <= forecast_horizon
+    X_separated_by_gap = are_datasets_separated_by_gap_time_index(
+        X_train, X, pipeline_params
+    )
+    if not (right_length and X_separated_by_gap):
+        raise PartialDependenceError(
```

Comment:
Based on the second point of [this](https://github.com/alteryx/evalml/issues/3078#issuecomment-1010307947): This helper function will return "something" if the test data violates our constraints. Users can then use this "something" to display warning messages prior to calling predict.

Reply:
i think that makes sense